scripts/verify-release.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+
+RELEASE_TAG="$1"
+
+GITHUB_TOKEN="${GITHUB_TOKEN:-}"
+OWNER="${OWNER:-openbao}"
+REPO="${REPO:-openbao}"
+TMPDIR="${TMPDIR:-/tmp}"
+GPG_KEY_NAME="${GPG_KEY_NAME:-openbao-gpg-pub-20240618.asc}"
+
+DOWNLOAD_FOLDER="${TMPDIR}/${OWNER}-${REPO}-${RELEASE_TAG}"
+
+if [ ! -e "$DOWNLOAD_FOLDER" ]; then
+  mkdir -p "$DOWNLOAD_FOLDER"
+fi
+
+release_cache="${TMPDIR}/${OWNER}-${REPO}-release-${RELEASE_TAG}.json"
+release_api="https://api.github.com/repos/${OWNER}/${REPO}/releases/tags/${RELEASE_TAG}"
+bearer_header=""
+if [ -n "$GITHUB_TOKEN" ]; then
+  bearer_header="Authorization: Bearer $GITHUB_TOKEN"
+fi
+
+release_info=""
+if [ -e "$release_cache" ]; then
+  release_info="$(cat "$release_cache")"
+else
+  release_info="$(curl -sSL \
+    -H "Accept: application/vnd.github+json" \
+    -H "$bearer_header" \
+    -H "X-GitHub-Api-Version: 2022-11-28" \
+    "$release_api")"
+  echo "$release_info" > "$release_cache"
+fi
+
+assets_count="$(jq '.assets | length' <<< "$release_info")"
+
+for index in $(seq 0 $(( assets_count - 1 ))); do
+  download_url="$(jq -r ".assets[$index].browser_download_url" <<< "$release_info")"
+  download_file="$(basename "$download_url")"
+  download_path="${DOWNLOAD_FOLDER}/${download_file}"
+
+  if [ ! -e "$download_path" ]; then
+    echo "Downloading $download_file..."
+    curl -sSL "$download_url" --output "$download_path"
+    sleep 2
+  fi
+done
+
+if [ ! -e "${DOWNLOAD_FOLDER}/${GPG_KEY_NAME}" ]; then
+  echo "Missing GPG file ($GPG_KEY_NAME); upload to release and rerun" 1>&2
+  exit 1
+fi
+
+cd "$DOWNLOAD_FOLDER"
+
+(
+  set -euxo pipefail
```

Comment:
@chicks-net Ahh, I see. If I remove this line, I get a warning about the line above:

```
[cipherboy@x1c openbao]$ shellcheck scripts/verify-release.sh 

In scripts/verify-release.sh line 55:
cd "$DOWNLOAD_FOLDER"
^-------------------^ SC2164 (warning): Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

Did you mean: 
cd "$DOWNLOAD_FOLDER" || exit

For more information:
  https://www.shellcheck.net/wiki/SC2164 -- Use 'cd ... || exit' or 'cd ... |...
```

so I think shellcheck assumes it has global impact, rather than understanding it only affects execution within the subshell and only after it has been executed.